python/nano/src/bigdl/nano/tf/keras/inference/optimizer.py

```diff
@@ -0,0 +1,262 @@
+        # TODO: how to obtain input from output of training_data
+        input_sample = input_sample[:-1]
+
+        if isinstance(input_sample, (list, tuple)) and len(input_sample) == 1:
+            input_sample = input_sample[0]
+
+        st = time.perf_counter()
+        try:
+            if isinstance(input_sample, tf.Tensor):
+                model(input_sample)
+            else:
+                model(*input_sample)
+        except Exception:
+            invalidInputError(False,
+                              "training_data is incompatible with your model input.")
+        baseline_time = time.perf_counter() - st
+        if baseline_time > 0.1:  # 100ms
+            sample_size_for_pot = 15
+        else:
+            sample_size_for_pot = 100
+
+        print("==========================Start Optimization==========================")
+        start_time = time.perf_counter()
+        for idx, (method, available) in enumerate(available_dict.items()):
+            result_map[method] = {}
+            if available is False:
+                result_map[method]["status"] = "lack dependency"
+            else:
+                print(f"----------Start test {method} model "
+                      f"({idx+1}/{len(available_dict)})----------")
+                option: AccelerationOption = self.ALL_INFERENCE_ACCELERATION_METHOD[method]
+                precision: str = option.get_precision()
+                try:
+                    acce_model = option.optimize(model=model,
+                                                 training_data=training_data,
+                                                 input_sample=tf.TensorSpec(
+                                                     shape=input_sample.shape,
+                                                     dtype=tf.float32),
+                                                 thread_num=thread_num,
+                                                 logging=logging,
+                                                 sample_size_for_pot=sample_size_for_pot)
+                except Exception as e:
+                    print(e)
+                    result_map[method]["status"] = "fail to convert"
+                    print(f"----------Failed to convert to {method}----------")
+                    continue
+
+                result_map[method]["status"] = "successful"
+
+                def func_test(model, sample):
+                    model(sample)
+                try:
+                    result_map[method]["latency"], status =\
+                        throughput_calculate_helper(latency_sample_num, baseline_time,
+                                                    func_test, acce_model, input_sample)
+                    if status is False and method != "original":
+                        result_map[method]["status"] = "early stopped"
+                        continue
+                except Exception as e:
+                    result_map[method]["status"] = "fail to forward"
```

Comment:
nitpick: maybe also include the reason for failing

Reply:
done